lib/src/tasks/gen_test_runner/api.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library dart_dev.src.tasks.gen_test_runner.api;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:dart_dev/src/tasks/gen_test_runner/config.dart';
+import 'package:dart_dev/src/tasks/task.dart';
+
+Future<GenTestRunnerTask> genTestRunner(
+    {String directory,
+    Environment environment,
+    String filename,
+    bool genHtml,
+    bool react,
```

Comment:
I'm a little iffy about including this `react` flag in the API - it makes quite a few assumptions and is pretty specific to our uses. It assumes client-side react, but server-side react is valid. It's also tied to the `react_client.dart` import as well as the `setClientConfiguration()` api - if either of those were to change, it would put this task in the awkward position of having to support old and new.

I think we could facilitate the usage of react since it will be common for us without coupling it so closely to this API by providing a ReactClientHelper constant (that could be versioned to handle breaking changes) or something that would serve the purpose of injecting the react import and the configuration call.